trappy/ftrace.py

```diff
@@ -88,6 +145,13 @@ def __init__(self, name="", normalize_time=True, scope="all",
         if normalize_time:
             self.normalize_time()
 
+        if self.__class__.disable_cache: return
+
+        self._create_trace_cache()
+        for trace_class in self.trace_classes:
+            csv_file = self._get_csv_path(trace_class)
+            trace_class.write_csv(csv_file)
+
```

Comment:
[Continuing from [previous thread](https://github.com/ARM-software/trappy/pull/247#discussion_r118583125), which Github won't let me reply to any more]

> why this should be a try/except? If we fail to write to the cache for some reason, I'd rather treat it as a failure and consider it a bug. Are you worried about the system running out of space?

I was actually thinking of something like a permissions error - e.g. sometimes at ARM people share files via NFS mounts where only they have write permissions. Just in general the fact that you intuitively expect to be able to parse traces from read-only places. 

Maybe just catch `OSError`s?